apollo-federation/src/sources/connect/expand/mod.rs

```diff
@@ -614,6 +614,32 @@ mod helpers {
                         ))
                     }
                 }?;
+            } else {
+                // If we don't have a key, but the type is an interfaceObject
+                // we'll try just copying over the defined keys.
+                let output_ty = output_type.get(self.original_schema.schema())?;
+                let has_interface_object = output_ty
+                    .directives()
+                    .iter()
+                    .any(|d| d.name == name!("federation__interfaceObject"));
```

Comment:
I'm not positive how this works, is this name guaranteed, or can it be renamed? Should we instead be looking to see that the output type is an `interface` and has `join__type(isInterfaceObject: true)`?

Reply:
good question, i'll confirm.

Reply:
so expansion always happens on subgraphs extracted from the supergraph, which right now defines directives using [`new_empty_fed_2_subgraph_schema()` ](https://github.com/apollographql/router/blob/dev/apollo-federation/src/supergraph/schema.rs#L48) — there are no user-defined imports at this point.

but i should use constants regardless